tests/cover/test_fspaths.py

```diff
@@ -0,0 +1,218 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import io
+import os
+import sys
+import codecs
+
+import pytest
+
+from hypothesis import given, find
+from hypothesis.internal.compat import PY3, text_type
+from hypothesis.strategies import fspaths
+
+
+encoding = sys.getfilesystemencoding()
+is_win = (os.name == "nt")
+
+
+def norm_encoding(name):
+    """Normalizes an encoding name"""
+
+    return codecs.lookup(name).name
+
+
+def single_byte_full_encoding(encoding):
+    """Whether the encoding can decode all byte values (e.g. latin1)"""
+
+    if PY3:
+        bytes_ = map(lambda i: bytes([i]), range(0, 256))
+    else:
+        bytes_ = map(chr, range(0, 256))
+
+    for byte in bytes_:
+        try:
+            byte.decode(encoding)
+        except UnicodeDecodeError:
+            return False
+    return True
+
+
+def fspath(p):
+    """Like os.fspath but for Python <= 3.6"""
+
+    try:
+        return os.fspath(p)
+    except AttributeError:
+        return p
+
+
+@given(fspaths())
+def test_open(path):
+    # The value range of fspaths() is limited by what open() accepts
+    try:
+        with open(path):
```

Comment:
open defaults to mode="r", so there shouldn't be any side effects besides things like autofs doing mount on access, which wouldn't change when exists() is used as far as I can see.

I'll try to limit paths to a subdirectory in "/tmp"